spi/src/main/java/org/opensearch/security/spi/resources/ResourceAccessScope.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.security.spi.resources;
+
+import java.util.Arrays;
+
+/**
+ * This interface defines the two basic access scopes for resource-access. Plugins can decide whether to use these.
+ * Each plugin must implement their own scopes and manage them.
+ * These access scopes will then be used to verify the type of access being requested.
+ *
+ * @opensearch.experimental
+ */
+public interface ResourceAccessScope<T extends Enum<T>> {
+    String RESTRICTED = "restricted";
+    String PUBLIC = "public";
```

Comment:
NIT: These are not scopes, they are sharing levels

Reply:
These are scope names. I've updated one of them.